# Assessing Normal Distribution Activity

### Imports packages and loads the births14 dataset

### Column chosen is weight (birth weight in pounds)

In [7]:
import numpy as np
!pip install pandas
!pip install plotly
import pandas as pd
import plotly.express as px
import plotly.io as pio
pio.renderers.default = "iframe"
import requests
from io import StringIO

url = "https://www.openintro.org/data/csv/births14.csv"
headers = {'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/58.0.3029.110 Safari/537.3'}
response = requests.get(url, headers=headers)
response.raise_for_status() # Raise an exception for bad status codes
df = pd.read_csv(StringIO(response.text))

In [8]:
df

,fage,mage,mature,weeks,premie,visits,gained,weight,lowbirthweight,sex,habit,marital,whitemom
0,34.0,34,younger mom,37,full term,14.0,28.0,6.96,not low,male,nonsmoker,married,white
1,36.0,31,younger mom,41,full term,12.0,41.0,8.86,not low,female,nonsmoker,married,white
2,37.0,36,mature mom,37,full term,10.0,28.0,7.51,not low,female,nonsmoker,married,not white
3,NaN,16,younger mom,38,full term,NaN,29.0,6.19,not low,male,nonsmoker,not married,white
4,32.0,31,younger mom,36,premie,12.0,48.0,6.75,not low,female,nonsmoker,married,white
...,...,...,...,...,...,...,...,...,...,...,...,...,...
995,28.0,24,younger mom,39,full term,NaN,20.0,6.49,not low,male,NaN,not married,not white
996,37.0,31,younger mom,38,full term,8.0,33.0,5.80,not low,female,nonsmoker,married,white
997,27.0,27,younger mom,34,premie,7.0,25.0,6.75,not low,female,nonsmoker,married,white
998,NaN,33,younger mom,42,full term,0.0,13.0,7.44,not low,male,nonsmoker,not married,white


## Summary of data statistics for weight column

### The mean birth weight is around 7.2 lbs and the median is 7.31. The mean being slightly lower than the median (by 0.112 lbs) suggests that some lower birth weight babies (minimum is 0.75) are pulling down the mean (average) value, while the median (middle value) is lest impacted. This is a sign of a slight left skew in the babies birth weight.

In [26]:
print(df["weight"].describe())
mean = df["weight"].mean()
median = df["weight"].median()
print("Mean weight: ", mean, "lbs")
print("Median weight: ", median, "lbs")
print("Difference between mean and median: ", (mean - median), "lbs")

count    1000.000000
mean        7.198160
std         1.306775
min         0.750000
25%         6.545000
50%         7.310000
75%         8.000000
max        10.620000
Name: weight, dtype: float64
Mean weight:  7.19816 lbs
Median weight:  7.31 lbs
Difference between mean and median:  -0.11183999999999994 lbs


## Histogram of Weight Data

### The histogram shows a roughly bell-shaped distribution with a single peak around 7.5-8 lbs, which shows a with a normal distribution. However the left tail of the distribution is longer than the right. This is representing a small number of babies in the data were born under 3 lbs, while very few were born above 10 lbs on the upper end of the distribution. This confirms the slight left skew from noticing the mean was lower than the median in the summary of data. Overall the shape is approximately normal but not perfectly symmetric.


In [22]:
fig_histogram = px.histogram(
    df, 
    x="weight", 
    nbins=30, 
    title="Histogram of 1000 Baby Birth Weights", 
    labels={"weight": "Birth Weight (lbs)"},
)
fig_histogram.show()

## Boxplot of Weight Data

### The median (7.31 lbs) sits slightly above center of the box, Q1 is 6.54 and Q3 8.00, and the lower whisker down to 4.38 lb) is longer than the upper whisker at 10.13 lbs, also indicating slight left skew. The lower outlier dots between 4.38 lbs and 0.75 lbs likely represent premature births from the data. Overall the box plot reinforces the conclusion that birth weight is approximately normal but slightly left-skewed.

In [18]:
fig_box = px.box(
    df,
    y="weight",
    title="Box Plot of 1000 Baby Birth Weights",
    labels={"weight": "Birth Weight (lbs)"}
)
fig_box.show()

## Final Conclusion

### Based on both plots, baby birth weights from the data set are approximately normally distributed. The mean (7.20) and median (7.31) are close but not equal, the histogram shows a roughly bell-shaped peak, and the box plot shows roughly similar whisker lengths. The slight left skew caused by low birth weight outliers it is not perfectly normal.

### Final verdict: Yes, the birth weight data is approximately normally distributed, with a slight left skew caused by low weight outliers.